tools-local/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/RemoveInternetSourcesFromNuGetConfig.cs

```diff
@@ -67,6 +68,9 @@ public override bool Execute()
 
             packageSourcesElement.ReplaceNodes(local.ToArray());
 
+            // Remove disabledPackageSources element so if any internal packages remain, they are used in source-build
+            disabledPackageSourcesElement?.ReplaceNodes(new XElement("clear"));
```

Comment:
Hmm... do you know why this works for non-internal builds? I would expect it to try to use internal sources in the production build and fail, if we remove the disables unconditionally. It obviously didn't fail, but curious why.

Reply:
I don't.  My guess would be it's hitting the public feeds first, we've seen some nondeterministic behavior with which NuGet feed gets hit first before.  Think it's worth removing this?

Reply:
I think we should go ahead with and merge the PR, but also file an issue to look into it.

Reply:
Ahh right, poked around a bit, and we actually remove all internal sources all the time unless specifically included by specific projects when they need an internal prebuilt dependency like templates:

https://github.com/dotnet/source-build/blob/8cad9c79cb726997804c415782eaac07168e40d8/repos/core-sdk.proj#L77-L84

That only happens when auth is provided, so it works fine externally. Case closed. 😄

Reply:
Cool, thanks for looking into it!